test/test_tensor_creation_ops.py

```diff
@@ -708,51 +708,6 @@ def test_cat_different_devices(self, devices):
                                     "Expected all tensors to be on the same device"):
             torch.cat((cuda0, cuda0), out=cuda1)
 
-    @onlyCUDA
-    def test_cat_stack_cross_devices(self, device):
-        cuda = torch.randn((3, 3), device=device)
-        cpu = torch.randn((3, 3), device='cpu')
-        out_cpu = cpu.clone()
-        out_cuda = cuda.clone()
-        with self.assertRaisesRegex(RuntimeError,
-                                    "Expected all tensors to be on the same device"):
-            torch.cat((cuda, cpu))
-        with self.assertRaisesRegex(RuntimeError,
-                                    "Expected all tensors to be on the same device"):
-            torch.cat((cpu, cuda))
-
-        with self.assertRaisesRegex(RuntimeError,
-                                    "Expected all tensors to be on the same device"):
-            torch.cat((cpu, cuda), out=out_cuda)
-
-        with self.assertRaisesRegex(RuntimeError,
-                                    "Expected all tensors to be on the same device"):
-            torch.cat((cpu, cpu), out=out_cuda)
-
-        with self.assertRaisesRegex(RuntimeError,
-                                    "Expected all tensors to be on the same device"):
-            torch.cat((cuda, cuda), out=out_cpu)
-
-        # Stack
-        with self.assertRaisesRegex(RuntimeError,
-                                    "Expected all tensors to be on the same device"):
-            torch.stack((cuda, cpu))
-        with self.assertRaisesRegex(RuntimeError,
-                                    "Expected all tensors to be on the same device"):
-            torch.stack((cpu, cuda))
-
-        with self.assertRaisesRegex(RuntimeError,
-                                    "Expected all tensors to be on the same device"):
-            torch.stack((cpu, cuda), out=out_cuda)
-
-        with self.assertRaisesRegex(RuntimeError,
-                                    "Expected all tensors to be on the same device"):
-            torch.stack((cpu, cpu), out=out_cuda)
-
-        with self.assertRaisesRegex(RuntimeError,
-                                    "Expected all tensors to be on the same device"):
-            torch.stack((cuda, cuda), out=out_cpu)
```

Comment:
I'm not sure if we should remove these tests. If I'm not wrong, OpInfos don't test for cross-device inputs. (I may be wrong :)). cc-ing @kshitij12345

Reply:
https://github.com/pytorch/pytorch/blob/50fc8e8250adafb564e59dc90b39aca6d9b2cee9/test/test_ops.py#L174-L179

https://github.com/pytorch/pytorch/blob/50fc8e8250adafb564e59dc90b39aca6d9b2cee9/test/test_ops.py#L340-L354

IIRC, once `test_out` was added plan was to add OpInfo for `cat` and remove this redundant test. (But we might have to check that all cases are covered). But I am certain that there are few redundant cases.

cc: @mruberry

Reply:
I don't think that the tests you shared check for cases with multiple devices for `inputs`. You are right, there are a few redundant cases, so we can keep those that are not covered by OpInfos.

Example:

```python
torch.cat((cpu_input, cuda_input))
# or
torch.stack((cpu_input, cuda_input))
```

aren't checked.

Reply:
Nice catch @krshrimali! I'll add back the cases which are not covered in OpInfo tests.

Reply:
@kshitij12345 @krshrimali I've removed the redundant ones and added back the cross device ones in this [commit](https://github.com/pytorch/pytorch/pull/62560/commits/5b54571d2254f59959bbe6beea4ec1a0b39e08aa)